src/dftbp/dftbplus/initprogram.F90

```diff
@@ -1785,34 +1815,63 @@ subroutine initProgramVariables(this, input, env)
     ! Initialize mixer
     ! (at the moment, the mixer does not need to know about the size of the vector to mix.)
     if (this%tSccCalc .and. .not. allocated(this%reks) .and. .not. this%tRestartNoSC) then
-      allocate(this%pChrgMixer)
       iMixer = input%ctrl%iMixSwitch
       nGeneration = input%ctrl%iGenerations
       mixParam = input%ctrl%almix
+      if (this%isHybridXc) then
+        if ((.not. this%tRealHS) .and. (input%ctrl%hybridXcInp%hybridXcAlg&
```

Comment:
```suggestion
        tCmplxMixer = (.not. this%tRealHS) .and. (input%ctrl%hybridXcInp%hybridXcAlg == hybridXcAlgo%matrixBased)
```
Might be clearer

Reply:
Yes, but had to switch `input%ctrl%hybridXcInp%hybridXcAlg` $\rightarrow$ `this%hybridXcAlg`, since this would otherwise result in segfaults for non-hybrid runs.